packages/clerk-js/src/core/clerk.ts

```diff
@@ -1003,6 +1010,27 @@ export class Clerk implements ClerkInterface {
     await onAfterSetActive();
   };
 
+  #handlePendingSession = async (session: PendingSessionResource) => {
+    if (!session.currentTask || !this.environment) {
+      return;
+    }
+
+    if (session?.lastActiveToken) {
+      eventBus.dispatch(events.TokenUpdate, { token: session.lastActiveToken });
+    }
+
+    if (this.#internalComponentNavigate) {
+      // Handles navigation for UI components
+      await this.#internalComponentNavigate(session.currentTask.__internal_getPath());
+    } else {
+      // Handles navigation for custom flows
+      await this.navigate(session.currentTask.__internal_getUrl(this.#options, this.environment));
+    }
+
+    this.#setAccessors(session);
+    this.#emit();
```

Comment:
Our Dashboard contains the following snippet: 

```
<div className='relative z-20 [--animation-duration:250ms] [&>div]:animate-fade-in'>
                <SignIn
                  routing='path'
                  path='/sign-in'
```

The below animation gets triggered again and it causes this small flickering on navigation: 


https://github.com/user-attachments/assets/04d6fedc-3899-40bd-93bf-167e7ae5d25e


`this.#emit` updates `Clerk.session` which causes all React tree under `MultisessionAppSupport` to re-render, leading to a DOM re-painting and re-triggering the animation 

I think this is an issue unrelated to the current PR, and we didn't see it before since the previous sign-in routes didn't rely on `Clerk.session`